Unable to build using clang

```
cc    -c -o budpop.o budpop.c
budpop.c:56:3: warning: add explicit braces to avoid dangling else [-Wdangling-else]
                else
                ^
budpop.c:63:32: warning: inequality comparison result unused [-Wunused-comparison]
        for (ai = res; ai != NULL; ai != ai->ai_next) {
                                   ~~~^~~~~~~~~~~~~~
budpop.c:63:32: note: use '|=' to turn this inequality comparison into an or-assignment
        for (ai = res; ai != NULL; ai != ai->ai_next) {
                                      ^~
                                      |=
budpop.c:91:2: error: expected expression
        fd_set rfds;
        ^
budpop.c:92:11: error: use of undeclared identifier 'rfds'
        FD_ZERO(&rfds);
                 ^
budpop.c:93:24: error: use of undeclared identifier 'rfds'
        FD_SET(STDIN_FILENO, &rfds);
                              ^
budpop.c:94:14: error: use of undeclared identifier 'rfds'
        FD_SET(fd, &rfds);
                    ^
budpop.c:98:20: error: use of undeclared identifier 'rfds'; did you mean 'nfds'?
        if (select(nfds, &rfds, NULL, NULL, NULL) == -1)
                          ^~~~
                          nfds
budpop.c:96:6: note: 'nfds' declared here
        int nfds;
            ^
budpop.c:106:20: error: use of undeclared identifier 'rfds'; did you mean 'nfds'?
        if (FD_ISSET(fd, &rfds)) {
                          ^~~~
                          nfds
/usr/include/x86_64-linux-gnu/sys/select.h:87:46: note: expanded from macro 'FD_ISSET'
#define FD_ISSET(fd, fdsetp)    __FD_ISSET (fd, fdsetp)
                                                ^
/usr/include/x86_64-linux-gnu/bits/select.h:37:17: note: expanded from macro '__FD_ISSET'
  ((__FDS_BITS (s)[__FD_ELT (d)] & __FD_MASK (d)) != 0)
                ^
/usr/include/x86_64-linux-gnu/sys/select.h:68:28: note: expanded from macro '__FDS_BITS'
# define __FDS_BITS(set) ((set)->__fds_bits)
                           ^
budpop.c:96:6: note: 'nfds' declared here
        int nfds;
            ^
budpop.c:106:6: error: member reference base type 'int' is not a structure or union
        if (FD_ISSET(fd, &rfds)) {
            ^~~~~~~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/sys/select.h:87:30: note: expanded from macro 'FD_ISSET'
#define FD_ISSET(fd, fdsetp)    __FD_ISSET (fd, fdsetp)
                                ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/bits/select.h:37:5: note: expanded from macro '__FD_ISSET'
  ((__FDS_BITS (s)[__FD_ELT (d)] & __FD_MASK (d)) != 0)
    ^~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/sys/select.h:68:32: note: expanded from macro '__FDS_BITS'
# define __FDS_BITS(set) ((set)->__fds_bits)
                          ~~~~~^ ~~~~~~~~~~
budpop.c:127:4: warning: add explicit braces to avoid dangling else [-Wdangling-else]
                        else
                        ^
3 warnings and 7 errors generated.
make: *** [<builtin>: budpop.o] Error 1
```

--- a/budpop.c
+++ b/budpop.c
@@ -60,7 +60,7 @@ main(int argc, char *argv[])
 	struct addrinfo *ai;
 	int fd;
 	const char *cause;
-	for (ai = res; ai != NULL; ai != ai->ai_next) {
+	for (ai = res; ai != NULL; ai = ai->ai_next) {
 		fd = socket(ai->ai_family, ai->ai_socktype, ai->ai_protocol);
 		if (fd == -1) {
 			cause = "socket";

--- a/budpop.c
+++ b/budpop.c
@@ -87,7 +87,7 @@ main(int argc, char *argv[])
 		err(EXIT_FAILURE, "%s", cause);
 	freeaddrinfo(res);
 
-loop:
+loop:;
 	fd_set rfds;
 	FD_ZERO(&rfds);
 	FD_SET(STDIN_FILENO, &rfds);

